src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,171 @@
+        if (index == 0) {
+            nodeToPut = new Node<>(index, key, value, null);
+        } else {
+            nodeToPut = new Node<>(key.hashCode(), key, value, null);
+        }
+
+        if (table[index] == null) {
+            table[index] = nodeToPut;
+            size++;
+            return;
+        }
+
+        Node<K, V> currentNode = table[index];
+        while (currentNode != null) {
+            if (Objects.equals(currentNode.key, key)) {
+                currentNode.value = value;
+                return;
+            }
+
+            if (currentNode.next == null) {
+                currentNode.next = nodeToPut;
+                size++;
+                return;
+            }
+            currentNode = currentNode.next;
+        }
 
     }
 
     @Override
     public V getValue(K key) {
+        int index = (key == null) ? 0 : Math.abs(key.hashCode() % table.length);
+        Node<K, V> currentNode = table[index];
+
+        while (currentNode != null) {
+            if (Objects.equals(currentNode.key, key)) {
+                return currentNode.value;
+            }
+            currentNode = currentNode.next;
+        }
+
         return null;
     }
 
     @Override
     public int getSize() {
-        return 0;
+        return size;
+    }
+
+    @Override
+    public boolean equals(Object obj) {
+        if (obj == null) {
+            return false;
+        }
+        
+        if (obj == this) {
+            return true;
+        }
+
+        if (obj.getClass() != getClass()) {
+            return false;
+        }
+
+        MyHashMap<K, V> other = (MyHashMap<K, V>) obj;
+
+        if (size != other.size) {
+            return false;
+        }
+
+        for (Node<K, V> node : table) {
+            while (node != null) {
+                Object otherValue = other.getValue(node.key);
+                if (!Objects.equals(node.value, otherValue)) {
+                    return false;
+                }
+                node = node.next;
+            }
+        }
+        return true;
+    }
+
+    @Override
+    public int hashCode() {
+        int hash = 0;
+        for (Node<K, V> node : table) {
+            while (node != null) {
+                hash += Objects.hashCode(node.key) ^ Objects.hashCode(node.value);
+                node = node.next;
+            }
+        }
+        return hash;
+    }
+
+    private void resize() {
+        Node<K, V>[] newTable = new Node[table.length * 2];
+
+        for (Node<K, V> node : table) {
+            while (node != null) {
+                int index = (node.key == null) ? 0
+                        : Math.abs(node.key.hashCode() % newTable.length);
+
+                Node<K, V> nextNode = node.next;
+                node.next = newTable[index];
+                newTable[index] = node;
+                node = nextNode;
+            }
+        }
+
+        table = newTable;
+        threshold = (int) (DEFAULT_LOAD_FACTOR * newTable.length);
+    }
```

Comment:
The `resize` method does not update the `size` variable. This is crucial for maintaining the correct state of the hashmap after resizing. Ensure that you update the `size` variable appropriately within this method.